test/Microsoft.AspNetCore.Server.KestrelTests/SocketInputTests.cs

```diff
@@ -1,243 +0,0 @@
-
-                await task2.ContinueWith(
-                    (t) =>
-                    {
-                        TestConcurrentFaultedTask(t);
-                        task2Threw = true;
-                    },
-                    TaskContinuationOptions.OnlyOnFaulted);
-
-                Assert.True(task2.IsFaulted);
-                Assert.True(task2Threw);
-            }
-        }
-
-        [Fact]
-        public void ConsumingOutOfOrderFailsGracefully()
-        {
-            var defultIter = new MemoryPoolIterator();
-
-            // Calling ConsumingComplete without a preceding calling to ConsumingStart fails
-            using (var socketInput = new SocketInput(null, null))
-            {
-                Assert.Throws<InvalidOperationException>(() => socketInput.ConsumingComplete(defultIter, defultIter));
-            }
-
-            // Calling ConsumingComplete twice in a row fails
-            using (var socketInput = new SocketInput(null, null))
-            {
-                socketInput.ConsumingStart();
-                socketInput.ConsumingComplete(defultIter, defultIter);
-                Assert.Throws<InvalidOperationException>(() => socketInput.ConsumingComplete(defultIter, defultIter));
-            }
-
-            // Calling ConsumingStart twice in a row fails
-            using (var socketInput = new SocketInput(null, null))
-            {
-                socketInput.ConsumingStart();
-                Assert.Throws<InvalidOperationException>(() => socketInput.ConsumingStart());
-            }
-        }
-
-        [Fact]
-        public async Task PeekAsyncRereturnsTheSameData()
-        {
-            using (var memory = new MemoryPool())
-            using (var socketInput = new SocketInput(memory, new SynchronousThreadPool()))
-            {
-                socketInput.IncomingData(new byte[5], 0, 5);
-
-                Assert.True(socketInput.IsCompleted);
-                Assert.Equal(5, (await socketInput.PeekAsync()).Count);
-
-                // The same 5 bytes will be returned again since it hasn't been consumed.
-                Assert.True(socketInput.IsCompleted);
-                Assert.Equal(5, (await socketInput.PeekAsync()).Count);
-
-                var scan = socketInput.ConsumingStart();
-                scan.Skip(3);
-                socketInput.ConsumingComplete(scan, scan);
-
-                // The remaining 2 unconsumed bytes will be returned.
-                Assert.True(socketInput.IsCompleted);
-                Assert.Equal(2, (await socketInput.PeekAsync()).Count);
-
-                scan = socketInput.ConsumingStart();
-                scan.Skip(2);
-                socketInput.ConsumingComplete(scan, scan);
-
-                // Everything has been consume so socketInput is no longer in the completed state
-                Assert.False(socketInput.IsCompleted);
-            }
-        }
-
-        [Fact]
-        public async Task CompleteAwaitingDoesNotCauseZeroLengthRead()
```

Comment:
If there aren't tests like this in corefxlabs already, it might be worth copying.